app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -823,22 +790,16 @@ export function testsForRpcMethodSupportingBlockParam(
         comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
         // The block-ref middleware will make the request as specified
         // except that the block param is replaced with the latest block
-        // number.
+        // number, and we delay it.
         comms.mockInfuraRpcCall({
+          delay: 100,
```

Comment:
@mcmire Let me know if these test changes make sense! I did one read-through and removed any mocks that seemed no longer necessary. It's possible I missed some though - extra mocks don't cause tests to fail at the moment.

This line was the one I was least sure about. I didn't know for sure whether adding a delay to the _child_ request sufficiently demonstrates the behaviour we were looking for here or not.

Reply:
Gotcha. I think that's fine to remove the extra mocks.

In this case, though, I think it's important that we delay one of these requests to prevent this test from failing intermittently. This test is different from other tests in that we issue multiple requests at the exact same time. We want the first one (the one with the first result) to be processed first, and then the other ones afterward. So the delay should guarantee the right result.

Reply:
Right. So this line is a delay on the first request. I did this because the original test included a delay on the first "real" request, the one that would overwrite this one. Since that request no longer happens, I moved it to this request, which is now the first request.

Reply:
Ah yes, okay. I think I had trouble reading the diff. This looks good!